src/scripts/main.js

```diff
@@ -1,3 +1,269 @@
 'use strict';
 
-// write code here
+const thead = document.querySelector('thead').querySelector('tr');
+const tbody = document.querySelector('tbody');
+let workerData;
+const clickCounts = {};
+
+// #region sorting
+
+thead.addEventListener('click', (e) => {
+  workerData = [];
+
+  // eslint-disable-next-line max-len
+  const allWorkers = [...document.querySelector('tbody').querySelectorAll('tr')];
+  // eslint-disable-next-line max-len
+  const theadElements = [...document.querySelector('thead').querySelectorAll('th')];
+
+  collect(allWorkers, theadElements);
+
+  if (!clickCounts.hasOwnProperty(`${e.target.innerText}`)) {
+    clickCounts[e.target.innerText] = 0;
+  }
+
+  // eslint-disable-next-line max-len
+  const sortedData = sort(workerData, e.target.innerText, clickCounts[e.target.innerText]);
+
+  clickCounts[e.target.innerText]++;
+
+  for (const key in clickCounts) {
+    if (key !== e.target.innerText) {
+      clickCounts[key] = 0;
+    }
+  }
+
+  changeData(allWorkers, theadElements, sortedData);
+});
+
+function collect(allWorkers, theadElements) {
+  allWorkers.forEach((worker) => {
+    const workerInfo = {};
+
+    for (let j = 0; j < theadElements.length; j++) {
+      workerInfo[theadElements[j].innerText] = worker.children[j].innerText;
+    }
+
+    workerData.push(workerInfo);
+  });
+}
+
+function sort(list, sortLine, direction) {
+  switch (sortLine) {
+    case 'Name':
+    case 'Position':
+    case 'Office':
+      return sortString(list, sortLine, direction);
+    case 'Age':
+      return sortNumbers(list, sortLine, direction);
+    case 'Salary':
+      return sortSalary(list, sortLine, direction);
+  }
+}
+
+function sortSalary(list, sortLine, direction) {
+  return (direction % 2 === 0)
+    ? list
```

Comment:
avoid mutating lists in helper function (use only [pure functions](https://www.geeksforgeeks.org/pure-functions-in-javascript/))